Sources/Screen/VideoRotator.swift

```diff
@@ -0,0 +1,173 @@
+        }
+        switch orientation {
+        case .up, .down:
+            pixelInfo = .init(width: buffer.width, height: buffer.height, format: buffer.formatType)
+        case .left, .right:
+            pixelInfo = .init(width: buffer.height, height: buffer.width, format: buffer.formatType)
+        default:
+            return .failure(.unsupportedOrientation)
+        }
+        guard let rotationPixelBuffer else {
+            return .failure(.cannotAllocatePixelBuffer(pixelBufferStatus))
+        }
+        status = VTPixelRotationSessionRotateImage(session, buffer, rotationPixelBuffer)
+        guard status == noErr else {
+            return .failure(.rotationFailure(status))
+        }
+        var rotatedBuffer: CMSampleBuffer?
+        (rotatedBuffer, status) = createSampleBuffer(sampleBuffer, imageBuffer: rotationPixelBuffer)
+        guard let rotatedBuffer else {
+            return .failure(.rotationFailure(status))
+        }
+        return .success(rotatedBuffer)
+    }
+
+    @inline(__always)
+    private func createSampleBuffer(_ inSampleBuffer: CMSampleBuffer, imageBuffer: CVImageBuffer) -> (CMSampleBuffer?, OSStatus) {
+        var info = CMSampleTimingInfo()
+        info.presentationTimeStamp = inSampleBuffer.presentationTimeStamp
+        info.duration = CMSampleBufferGetOutputDuration(inSampleBuffer)
+        info.decodeTimeStamp = CMSampleBufferGetDecodeTimeStamp(inSampleBuffer)
+        var formatDescription: CMFormatDescription?
+        var status = CMVideoFormatDescriptionCreateForImageBuffer(
+            allocator: kCFAllocatorDefault,
+            imageBuffer: imageBuffer,
+            formatDescriptionOut: &formatDescription)
+        guard status == noErr, let formatDescription else {
+            return (nil, status)
+        }
+        var outSampleBuffer: CMSampleBuffer?
+        status = CMSampleBufferCreateReadyWithImageBuffer(allocator: kCFAllocatorDefault,
+                                                          imageBuffer: imageBuffer,
+                                                          formatDescription: formatDescription,
+                                                          sampleTiming: &info,
+                                                          sampleBufferOut: &outSampleBuffer)
+
+        guard status == noErr, let outSampleBuffer else {
+            return (nil, status)
+        }
+        return (outSampleBuffer, noErr)
+    }
+}
+
+private struct PixelInfo: Hashable, Equatable, CustomStringConvertible {
+    static let zero: PixelInfo = .init(width: 0, height: 0, format: 0)
+
+    let width: Int
+    let height: Int
+    let format: OSType
+
+    static func == (lhs: PixelInfo, rhs: PixelInfo) -> Bool {
+        return lhs.width == rhs.width && lhs.height == rhs.height && lhs.format == rhs.format
+    }
+
+    func hash(into hasher: inout Hasher) {
+        hasher.combine(width)
+        hasher.combine(height)
+        hasher.combine(format)
+    }
```

Comment:
[IMO]
- Let's remove it. If it's a private struct, adding Hashable and Equatable will automatically generate the relevant code. Let Xcode handle it instead of writing it yourself.